src/bin/languageserver/mod.rs

```diff
@@ -178,22 +243,31 @@ impl SolangServer {
 
             let res = self.client.publish_diagnostics(uri, diags, None);
 
-            let hovers = Builder::build(&ns);
+            let (caches, definitions) = Builder::build(&ns);
 
-            self.files.lock().await.hovers.insert(path, hovers);
+            let mut files = self.files.lock().await;
+            for (f, c) in ns.files.iter().zip(caches.into_iter()) {
+                if f.cache_no.is_some() {
+                    files.caches.insert(f.path.clone(), c);
+                }
+            }
+
+            self.definitions.lock().await.extend(definitions);
 
             res.await;
         }
     }
 }
 
 struct Builder<'a> {
-    hovers: Vec<HoverEntry>,
+    hovers: Vec<(usize, HoverEntry)>,
+    definitions: Definitions,
+    references: Vec<(usize, ReferenceEntry)>,
```

Comment:
Please, add a comment explaining what the `usize` means in hovers and references.